accepted/future-releases/inline-classes/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+(and there may also be some extension members).
+
+Hence, using a supertype as the static type allows us to see only a
+subset of the members. Using an inline type allows us to _replace_ the
+set of members, with subsetting as a special case.
+
+This functionality is entirely static. Invocation of an inline member is
+resolved at compile-time, based on the static type of the receiver.
+
+An inline class may be considered to be a zero-cost abstraction in the
+sense that it works similarly to a wrapper object that holds the
+wrapped object in a final instance variable. The inline class thus
+provides an interface which is chosen independently of the interface
+of the wrapped object, and it provides implementations of the members
+of the interface of the inline class, and those implementations can use
+the members of the wrapped object as needed.
+
+However, even though an inline class behaves like a wrapping, the wrapper
+object will never exist at run time, and a reference whose type is the
+inline class will actually refer directly to the underlying wrapped
+object. Every member access (e.g., an invocation of a method or a
+getter) on an expression whose static type is an inline type will invoke
+a member of the inline class (with some exceptions, as explained below),
+but this occurs because those member accesses are resolved statically,
+which means that the wrapper object is not actually needed.
+
+Given that there is no wrapper object, we will refer to the "wrapped"
+object as the _representation object_ of the inline class, or just the
+_representation_.
+
+Inside the inline class declaration, the keyword `this` is a reference
+to the representation whose static type is the enclosing inline
+class. A member access to a member of the enclosing inline class may
+rely on `this` being induced implicitly (for example, `foo()` means
+`this.foo()` if the inline class contains a method declaration named
+`foo`).
+
+A reference to the representation typed by its run-time type or a
+supertype thereof (that is, typed by a "normal" type for the
+representation) is available as a declared name: The inline class must
+have exactly one instance variable whose type is the representation
+type, and it must be `final`.
+
+The representation type of the inline class (with `final int i` that's
+`int`) is similar to the on-type of an extension declaration.
+
+All in all, an inline class allows us to replace the interface of a given
+representation object and specify how to implement the new interface
+in terms of the interface of the representation object.
+
+This is something that we could obviously do with a regular class used
+as a wrapper, but when it is done with an inline class there is no
+wrapper object, and hence there is no run-time performance cost. In
+particular, in the case where we have an inline type `V` with
+representation type `R` we may be able to refer to a `List<R>` using
```

Comment:
"may be able" sounds like it implies that we sometimes *won't* be able to refer to a `List<R>` using the type `List<V>`.  Is that true?  What are the circumstances in which we can't?